actors/builtin/miner/miner_actor.go

```diff
@@ -1375,18 +1399,27 @@ func (a Actor) WithdrawBalance(rt Runtime, params *WithdrawBalanceParams) *adt.E
 		if err != nil {
 			rt.Abortf(exitcode.ErrIllegalState, "failed to vest fund: %v", err)
 		}
+		unlockedBalance = st.GetUnlockedBalance(rt.CurrentBalance())
 
-		// Verify InitialPledgeRequirement does not exceed unlocked funds
-		verifyPledgeMeetsInitialRequirements(rt, &st)
+		// Verify unlocked funds cover both InitialPledgeRequirement and FeeDebt
+		// and repay fee debt now.
+		debtToBurn = verifyPledgeRequirementsAndRepayDebts(rt, &st)
+		unlockedBalance = big.Sub(unlockedBalance, debtToBurn)
```

Comment:
> It should return it.

I landed with this while fostering simplicity with the fund accounting because it is getting stressfully complicated and adding another return value to the interface gives me another calculation I need to rely on happening correctly in a subcall that I would rather see here directly.

Reply:
Well.. you're already relying on that calculation happening correctly in that subcall You just don't see it.